data/actions/scripts/other/potions.lua

```diff
@@ -1,158 +1,149 @@
-local ultimateHealthPot = 8473
-local greatHealthPot = 7591
-local greatManaPot = 7590
-local greatSpiritPot = 8472
-local strongHealthPot = 7588
-local strongManaPot = 7589
-local healthPot = 7618
-local manaPot = 7620
-local smallHealthPot = 8704
-local antidotePot = 8474
-local greatEmptyPot = 7635
-local strongEmptyPot = 7634
-local emptyPot = 7636
-
+--- CONFIGURATION ---
+potions = {
+	removePots = true,
+	addEmptyPots = true, -- Only works if removepots isn't false or nil
+	words = "Aaaah...",
+	[8473] = { -- ultimate health potion
+		vocations = {4, 8},
+		minLevel = 130,
+		health = { min = 650, max = 850},
+		empty = 7635
+	},
+	[7591] = { -- great health potion
+		vocations = {4, 8},
+		minLevel = 80,
+		health = { min = 425, max = 575},
+		empty = 7635
+	},
+	[7590] = { -- great mana potion
+		vocations = {1, 2, 5, 6},
+		minLevel = 80,
+		mana = { min = 150, max = 250},
+		empty = 7635
+	},
+	[8472] = { -- great spitit potion
+		vocations = {3, 7},
+		minLevel = 80,
+		health = { min = 250, max = 350},
+		mana = { min = 100, max = 200},
+		empty = 7635
+	},
+	[7588] = { -- strong health potion
+		vocations = {3, 4, 7, 8},
+		minLevel = 50,
+		health = { min = 250, max = 350},
+		empty = 7634
+	},
+	[7589] = { -- strong mana potion
+		vocations = {1, 2, 3, 5, 6, 7},
+		minLevel = 50,
+		mana = { min = 115, max = 185},
+		empty = 7634
+	},
+	[7620] = { -- mana potion
+		mana = { min = 75, max = 125},
+		empty = 7636
+	},
+	[7618] = { -- health potion
+		health = { min = 125, max = 175},
+		empty = 7636
+	},
+	[8704] = { -- small health potion
+		health = { min = 60, max = 90},
+		empty = 7636
+	}
+}
+-- antidote
 local antidote = Combat()
 antidote:setParameter(COMBAT_PARAM_TYPE, COMBAT_HEALING)
 antidote:setParameter(COMBAT_PARAM_EFFECT, CONST_ME_MAGIC_BLUE)
 antidote:setParameter(COMBAT_PARAM_TARGETCASTERORTOPMOST, true)
 antidote:setParameter(COMBAT_PARAM_AGGRESSIVE, false)
 antidote:setParameter(COMBAT_PARAM_DISPEL, CONDITION_POISON)
+potions[8474] = {combat = antidote, empty = 7636}
+
+--- FUNCTIONS ---
 
 local exhaust = Condition(CONDITION_EXHAUST_HEAL)
 exhaust:setParameter(CONDITION_PARAM_TICKS, (configManager.getNumber(configKeys.EX_ACTIONS_DELAY_INTERVAL) - 100))
 -- 1000 - 100 due to exact condition timing. -100 doesn't hurt us, and players don't have reminding ~50ms exhaustion.
 
-function onUse(player, item, fromPosition, target, toPosition, isHotkey)
+function onUse(player, potion, fromPosition, target, toPosition, isHotkey)
```

Comment:
> Well, there was a reason renaming it to potion, so renaming the missing item to potion is as well minor fix!

> A better name for this could be: potionId or potion.
I mean it is easier to understand what stats mean by looking at the right side of its assignment.

I guess you misunderstood it. I refered to this variable ``stats``:
```lua
local stats = potions[item:getId()]
```

You should rename ***stats*** to either ***potion*** or ***potionId***.
So it becomes:
```lua
local potion = potions[item:getId()]
```
I don't think the name **stats** makes any sense for this variable.